src/platform/OpenThread/GenericThreadStackManagerImpl_OpenThread.cpp

```diff
@@ -992,6 +998,237 @@ CHIP_ERROR GenericThreadStackManagerImpl_OpenThread<ImplClass>::_JoinerStart(voi
+
+    return MapOpenThreadError(error);
+}
+
+template <class ImplClass>
+CHIP_ERROR GenericThreadStackManagerImpl_OpenThread<ImplClass>::_StartSrpClient(const char * aHostName)
+{
+    otError error = OT_ERROR_NONE;
+    otIp6Address clientIpAddress;
+    otServiceConfig serverServiceConfig;
+
+    Impl()->LockThreadStack();
+
+    VerifyOrExit(aHostName, error = OT_ERROR_INVALID_ARGS);
+    VerifyOrExit(!otSrpClientIsRunning(mOTInst), error = OT_ERROR_ALREADY);
+
+    memcpy(&clientIpAddress, otThreadGetMeshLocalEid(mOTInst), sizeof(clientIpAddress));
+
+    error = otSrpClientSetHostName(mOTInst, aHostName);
+    VerifyOrExit(error == OT_ERROR_NONE, );
+
+    error = otSrpClientSetHostAddresses(mOTInst, &clientIpAddress, 1);
+    VerifyOrExit(error == OT_ERROR_NONE, );
+
+    VerifyOrExit(FindSrpServerService(serverServiceConfig), error = OT_ERROR_NOT_FOUND);
+
+    // Convert server service config data to get from it server IP address and port number.
+    otSockAddr serverSocketAddress;
+    serverSocketAddress.mAddress = ConvertSrpServerRloc16ToIpAddress(serverServiceConfig.mServerConfig.mRloc16);
+    serverSocketAddress.mPort    = ConvertSrpServerDataToPort(serverServiceConfig.mServerConfig.mServerData);
+
+    error = otSrpClientStart(mOTInst, &serverSocketAddress);
+
+exit:
+    Impl()->UnlockThreadStack();
+
+    return MapOpenThreadError(error);
+}
+
+template <class ImplClass>
+bool GenericThreadStackManagerImpl_OpenThread<ImplClass>::FindSrpServerService(otServiceConfig & aServiceConfig)
+{
+    bool serviceFound = false;
+
+    otNetworkDataIterator iterator = OT_NETWORK_DATA_ITERATOR_INIT;
+
+    // Find in the server services the one used for SRP server purposes
+    while (otNetDataGetNextService(mOTInst, &iterator, &aServiceConfig) == OT_ERROR_NONE)
+    {
+        if (aServiceConfig.mEnterpriseNumber == kThreadServiceEnterpriseNumber &&
+            aServiceConfig.mServiceDataLength == sizeof(kSrpServiceId) && aServiceConfig.mServiceData[0] == kSrpServiceId)
+        {
+            serviceFound = true;
+            break;
+        }
+    }
+
+    return serviceFound;
+}
+
+template <class ImplClass>
+uint16_t GenericThreadStackManagerImpl_OpenThread<ImplClass>::ConvertSrpServerDataToPort(uint8_t * aServerData)
+{
+    static_assert(OT_SERVER_DATA_MAX_SIZE >= 2, "");
+
+    // Swap server data bytes order
+    return static_cast<uint16_t>((aServerData[0] << 8u) + aServerData[1]);
+}
+
+template <class ImplClass>
+otIp6Address GenericThreadStackManagerImpl_OpenThread<ImplClass>::ConvertSrpServerRloc16ToIpAddress(uint16_t aRloc16)
+{
+    // Get current device address
+    otIp6Address serverIpAddress;
+    memcpy(&serverIpAddress, otThreadGetRloc(mOTInst), sizeof(serverIpAddress));
+
+    // Replace device rloc16 bytes with the server rloc16
+    uint16_t addrSize                        = sizeof(serverIpAddress.mFields.m8);
+    serverIpAddress.mFields.m8[addrSize - 2] = static_cast<uint8_t>((aRloc16 & 0xff00) >> 8u);
```

Comment:
Wouldn't something like `serverIpAddress.mFields.m16[7] = aRloc16` be better, alternatively if endianess would be different, don't we have sth like NTOHS or HostSwap16() function in CHIP..?